zcash_test_vectors/transparent/bip_0032.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+import sys; assert sys.version_info[0] >= 3, "Python 3 required."
+
+import hashlib
+import hmac
+from secp256k1 import PrivateKey, PublicKey
+
+from .zip_0316 import derive_ovks
+
+from ..hd_common import ZCASH_MAIN_COINTYPE, hardened
+from ..output import render_args, render_tv
+from ..utils import i2leosp
+
+
+class ExtendedSecretKey:
+    def __init__(self, sk, chaincode):
+        assert isinstance(sk, PrivateKey)
+        assert len(chaincode) == 32
+        self.sk = sk
+        self.chaincode = chaincode
+
+    @classmethod
+    def master(cls, S):
+        I = hmac.digest(b'Bitcoin seed', S, 'sha512')
+        I_L = I[:32]
+        I_R = I[32:]
+        sk = PrivateKey(I_L, True)
+        return cls(sk, I_R)
+
+    def __bytes__(self):
+        return self.chaincode + self.sk.private_key
+
+    def public_key(self):
+        return ExtendedPublicKey(self.sk.pubkey, self.chaincode)
+
+    def child(self, i):
+        assert 0 <= i and i <= 0xFFFFFFFF
+
+        if i >= 0x80000000:
+            I = hmac.digest(self.chaincode, b'\x00' + self.sk.private_key + i2leosp(32, i), 'sha512')
+        else:
+            I = hmac.digest(self.chaincode, self.sk.pubkey.serialize(compressed=True) + i2leosp(32, i), 'sha512')
```

Comment:
I was just checking over this again, and I realized I'm not sure whether this conforms to:
> If not (normal child): let I = HMAC-SHA512(Key = cpar, Data = serP(point(kpar)) || ser32(i)).

Does pubkey.serialize do:
> point(p): returns the coordinate pair resulting from EC point multiplication (repeated application of the EC group operation) of the secp256k1 base point with the integer p.

Reply:
https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki contains some test vectors, perhaps we should import those and use them to verify our bip32 implementation.

Reply:
Yes, that's what it does. And that's a good idea (but doesn't affect this PR).